lte/gateway/c/core/oai/test/ngap/test_ngap_flows.cpp

```diff
@@ -1413,4 +1413,28 @@ TEST_F(NgapFlowTest, test_ue_notifications_from_amf) {
   bdestroy(ngap_initial_ue_msg);
 }
 
+TEST_F(NgapFlowTest, NgapHandleSctpDisconnection) {
+  // Verify sctp association is successful
+  EXPECT_EQ(ngap_handle_new_association(state, &peerInfo), RETURNok);
+  // Verify number of connected gNB's is 1
+  EXPECT_EQ(state->gnbs.num_elements, 1);
+
+  // To test SCTP Shutdown for NGAP
+  bool reset = false;
+
+  gnb_description_t* gnb_association = NULL;
+  gnb_association = ngap_state_get_gnb(state, peerInfo.assoc_id);
+  ASSERT_TRUE(gnb_association != NULL);
+  EXPECT_EQ(gnb_association->ng_state, NGAP_INIT);
+  gnb_association->nb_ue_associated = 1;
```

Comment:
Instead of hard coding nb_ue_associated to 1, we can send "Registration Req" in Initial_ue_message. So that we can check the sending of de-registration indication to AMF app.
You can mock amf_app task. 
You can check number of active gNBs, after handling UE context release command.

Reply:
- Sent "Registration Req" in initial_ue_message, checked the ``NGAP_INITIAL_UE_MESSAGE`` messaged sent to amf_task. 
- Mocked the AMF_APP_NGAP_AMF_UE_ID_NOTIFICATION from AMF.
- Mocked the SCTP_CLOSE_ASSOCIATION ITTI message, checked the ``NGAP_GNB_DEREGISTERED_IND`` message sending to amf_task.
-  This test case ``TEST_F(AMFAppProcedureTest, SctpShutWithServiceRequest)`` in test_amf_procedures.cpp has amf side of sctp shutdown.
![image](https://user-images.githubusercontent.com/89978170/161259008-62769b9c-5c02-43f8-b6c9-e57849454e79.png)